engine-comparison/dispatcher.sh

```diff
@@ -7,60 +7,84 @@
   mkdir $ENGINE_DIR
 
   # Fuzzing activities go here
-
 }
 
 build_benchmark_using() {
   BENCHMARK=$1
   FENGINE_CONFIG=$2
+  THIS_BENCHMARK=$3
 
+  BUILDING_DIR=$WORK/BUILD-${THIS_BENCHMARK}
   echo "Filling $BUILDING_DIR"
-  BUILDING_DIR=$WORK/BUILD-${BENCHMARK}-WITH-${FENGINE_CONFIG}
   rm -rf $BUILDING_DIR
   mkdir $BUILDING_DIR
 
   # Do fuzzing things here
   # [[ ! -e ~/FTS/$BENCHMARK/build.sh ]] && echo "cant build" && exit 1
   # $WORK/FTS/$BENCHMARK/build.sh $FUZZING_ENGINE
 
-  export SEND_DIR=$WORK/SEND-${BENCHMARK}-WITH-${FENGINE_CONFIG}
+  export SEND_DIR=$WORK/SEND-${THIS_BENCHMARK}
   rm -rf $SEND_DIR
   mkdir $SEND_DIR
+  echo "Test file" > $SEND_DIR/example.txt
   # Then construct the directory to send here, e.g.
   # cp ${BENCHMARK}-${FUZZING_ENGINE} $SEND_DIR
   # copy seeds, afl-fuzz, etc
 
   rm -rf $BUILDING_DIR
 }
 
-BASE_INSTANCE_NAME="FTS-RUNNER"
+handle_benchmark() {
+  BENCHMARK=$1
+  FENGINE_CONFIG=$2
+  THIS_BENCHMARK=${BENCHMARK}-with-$(basename ${FENGINE_CONFIG}) # Just for convenience
 
-for FENGINE_CONFIG in $(find $WORK/fengine-configs); do
-  build_engine $FENGINE_CONFIG
-  for BENCHMARK in $ALL_BENCHMARKS; do
-    # n.b. this requires each config file to have a different name
-    build_benchmark_using $BENCHMARK $FENGINE_CONFIG
+  build_benchmark_using $BENCHMARK $FENGINE_CONFIG $THIS_BENCHMARK # & # ?
+  gsutil rsync -r $SEND_DIR gs://fuzzer-test-suite/binary-folders/
+  # GCloud instance names have lowercase restrictions
+  INSTANCE_NAME=$(echo "fts-runner-${THIS_BENCHMARK}" | tr '[:upper:]' '[:lower:]')
+  create_or_start $INSTANCE_NAME ./FTS/engine-comparison/runner-startup-script.sh
+}
+
+mkdir $WORK/fuzz-engines
 
-    THIS_BENCHMARK=${BENCHMARK}-WITH-${FENGINE_CONFIG} # Just for convenience
+# Stripped down equivalent of "gcloud init"
+gcloud auth activate-service-account $SERVICE_ACCOUNT \
+  --key-file="$WORK/FTS/engine-comparison/autogen/dispatcher-key.json"
+gcloud config set project fuzzer-test-suite
 
-    INSTANCE_NAME=${BASE_INSTANCE_NAME}-${THIS_BENCHMARK}
-    gcloud compute instances create $INSTANCE_NAME
 
-    gcloud compute scp --recurse $WORK/SEND-${THIS_BENCHMARK}/ ${INSTANCE_NAME}:/input
+# This config file defines $BMARKS
+. $WORK/FTS/engine-comparison/autogen/dispatcher.config
+# Now define $BENCHMARKS
+if [[ $BMARKS == 'all' ]]; then
+  for b in $(find ${SCRIPT_DIR}/../*/build.sh -type f); do
+    BENCHMARKS="$BENCHMARKS $(basename $(dirname $b))"
+  done
+elif [[ $BMARKS == 'small' ]]; then
+  BENCHMARKS="c-ares-CVE-2016-5180 re2-2014-12-09"
+#elif [[ $BMARKS == 'other alias' ]]; do
+else
+  BENCHMARKS=$(echo $1 | tr ',' ' ')
+fi
 
-    RUNNER_COMMAND="docker build --build-arg run-script=runner.sh /input"
-    gcloud compute ssh $INSTANCE_NAME --command=$RUNNER_COMMAND
+
+# Main working loops
+for FENGINE_CONFIG in $(find ${WORK}/fengine-configs/*); do
+  build_engine $FENGINE_CONFIG
+  for BENCHMARK in $BENCHMARKS; do
+    handle_benchmark $BENCHMARK $FENGINE_CONFIG # &
```

Comment:
Actually, it can't; not sure why, but I just tried it both ways and this singlehandedly breaks everything, as observed from the gcloud vm console